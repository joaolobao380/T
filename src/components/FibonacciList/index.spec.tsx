import { render } from '@testing-library/react-native';
import React from 'react';
import { useForm } from 'react-hook-form';

import { FibonacciSequence } from '.';

interface FarmWrapperProps {
  max: number;
  name: string;
}

const FormWrapper = ({ max, name }: FarmWrapperProps) => {
  const { control } = useForm({
    defaultValues: {
      [name]: null,
    },
  });

  return <FibonacciSequence control={control} max={max} name={name} label="Story points" />;
};

describe('FibonacciSequence Component', () => {
  it('renders correctly', () => {
    const max = 10;
    const name = 'fibonacciValue';
    const { getAllByText } = render(<FormWrapper max={max} name={name} />);
    expect(getAllByText('1').length).toBe(1);
    expect(getAllByText('2').length).toBe(1);
    expect(getAllByText('3').length).toBe(1);
    expect(getAllByText('5').length).toBe(1);
    expect(getAllByText('8').length).toBe(1);
  });

  it('Renders correctly Accordion', () => {
    const max = 10;
    const name = 'fibonacciValue';
    const { toJSON } = render(<FormWrapper max={max} name={name} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
