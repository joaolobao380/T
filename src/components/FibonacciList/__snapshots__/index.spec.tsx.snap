// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FibonacciSequence Component Renders correctly Accordion 1`] = `
<View>
  <View>
    <Text
      style={
        {
          "color": "#2A2A2A",
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      Story points
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 8,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#444444",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
            }
          }
        >
          1
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#DBDBDB",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000000",
            }
          }
        >
          2
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#DBDBDB",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000000",
            }
          }
        >
          3
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#DBDBDB",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000000",
            }
          }
        >
          5
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#DBDBDB",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "width": 40,
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#000000",
            }
          }
        >
          8
        </Text>
      </View>
    </View>
  </View>
</View>
`;
